src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -546,18 +548,16 @@ void WriteLog<I>::process_work() {
       this->m_wake_up_requested = false;
     }
     if (this->m_alloc_failed_since_retire || (this->m_shutting_down) ||
-        this->m_invalidating || m_bytes_allocated > high_water_bytes ||
-        (m_log_entries.size() > high_water_entries)) {
+        this->m_invalidating || m_bytes_allocated > high_water_bytes) {
       ldout(m_image_ctx.cct, 10) << "alloc_fail=" << this->m_alloc_failed_since_retire
                                  << ", allocated > high_water="
                                  << (m_bytes_allocated > high_water_bytes)
                                  << ", allocated_entries > high_water="
                                  << (m_log_entries.size() > high_water_entries)
                                  << dendl;
       retire_entries((this->m_shutting_down || this->m_invalidating ||
-                    (m_bytes_allocated > aggressive_high_water_bytes) ||
-                    (m_log_entries.size() > aggressive_high_water_entries))
-                    ? MAX_ALLOC_PER_TRANSACTION : MAX_FREE_PER_TRANSACTION);
+                     (m_bytes_allocated > aggressive_high_water_bytes))
```

Comment:
This fixes the bug where `retire_entries` is not triggered and it hangs when ssd cache is full. This is because the code was wrongly making a retire decision based on `m_log_entries`

Reply:
Actually `m_log_entries` size is irrelevant for SSD cache, therefore this check is is not meaningful and not necessary.